src/cleanup.rs

```diff
@@ -0,0 +1,168 @@
+                // The empty applyset must be mounted to be seen by the container.
+                "-v",
+                &format!("{}:{}", deletion_dir, deletion_dir),
+                "--env",
+                KUBECTL_APPLYSET_ENABLED,
+                "--env",
+                "KUBECONFIG=/.kube/config",
+                KUBECTL_IMAGE,
+            ]
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>(),
+        );
+    } else {
+        cli.extend(
+            ["kubectl"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>(),
+        );
+    }
+
+    cli.extend(
+        [
+            "apply",
+            "-n",
+            &app_instance.namespace_any(),
+            "--server-side",
+            "--prune",
+            "--applyset",
+            &app_instance.name_any(),
+            "--field-manager",
+            KUBIT_APPLIER_FIELD_MANAGER,
+            "--force-conflicts",
+            "-v=2",
+            "-f",
+            deletion_dir,
+        ]
+        .iter()
+        .map(|s| s.to_string())
+        .collect::<Vec<_>>(),
+    );
+
+    cli
+}
+
+pub fn emit_post_deletion_commandline(app_instance: &AppInstance, docker: bool) -> Vec<String> {
+    let mut cli: Vec<String> = vec![];
+
+    if docker {
+        let user_home = home_dir().expect("unable to retrieve home directory");
+        let kube_config =
+            env::var("KUBECONFIG").unwrap_or(format!("{}/.kube/config", user_home.display()));
+        cli.extend(
+            [
+                "docker",
+                "run",
+                "--interactive",
+                "--rm",
+                "--network",
+                "host",
+                "-v",
+                &format!("{}:/.kube/config", kube_config),
+                "--env",
+                "KUBECONFIG=/.kube/config",
+                KUBECTL_IMAGE,
+            ]
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>(),
+        );
+    } else {
+        cli.extend(
+            ["kubectl"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>(),
+        );
+    }
+
+    cli.extend(
+        [
+            "delete",
+            "configmap",
+            &configmap_name_for(app_instance.name_any()),
+            "--namespace",
+            &app_instance.namespace_any(),
+        ]
+        .iter()
+        .map(|s| s.to_string())
+        .collect::<Vec<_>>(),
+    );
+
+    cli
+}
+
+pub fn emit_deletion_setup(ns: &str, output_file: &str) -> Vec<String> {
+    [
+        "kubit",
+        "helper",
+        "cleanup",
+        "--namespace",
+        ns,
+        "--output",
+        output_file,
+    ]
+    .iter()
+    .map(|s| s.to_string())
+    .collect::<Vec<_>>()
+}
+
+/// Utility to generate the cleanup configmap name based on a given name.
+pub fn configmap_name_for(name: String) -> String {
```

Comment:
another nitpick: unlike Go in rust people can "use" a function from a modiule and thus reference a function without the name of the module where it's defined.

This means that a very generic name like `configmap_name_for` can be confusing if one doesn't have the immediate context that this is the "cleanup configmap".

Perhaps we can call it `pub fn cleanup_hack_resource_name(app_instance: &AppInstance) -> String`

?

Reply:
Yeah I can rename this :blush: 

I took the naming from the [`job_name_for`](https://github.com/kubecfg/kubit/blob/f66efe589b4032be894a3f0f172ceb863dcac2cb/src/controller.rs#L402C4-L404) func which already exists to keep it things consistently named.

---

Just so I understand your point, this is in the sense of it could be called like

```rust
use cleanup::configmap_name_for;

// more sweet rust code

configmap_name_for(...); // No module context on the call that this is for cleanup
```